v2.2/cost-based-optimizer.md

```diff
@@ -143,6 +143,25 @@ The query plan cache is still under development and has the following limitation
 - If you use the query plan cache in conjunction with table statistics, cached plans do not yet get invalidated when new statistics are created.
 {{site.data.alerts.end}}
 
+## Join reordering
+
+<span class="version-tag">New in v2.2</span>: The cost-based optimizer will explore additional join orderings in an attempt to find the lowest-cost execution plan for a query involving multiple joins. Specifically, it applies the rule of join associativity to generate even more possible join orderings to choose from, which can lead to significantly better performance in some cases.
+
+Because this process leads to an exponential increase in the number of possible execution plans for such queries, it's only used on queries with 4 or fewer joins by default, as determined by the `experimental_reorder_joins_limit` [session variable](set-vars.html).
```

Comment:
It's not really "queries with 4 or fewer joins", it's "will reorder subtrees containing 4 or fewer joins." But I'm not sure if that's difficult to communicate here?

The difference is that if a query has, say, 8 joins, the subtree it has containing 4 *will* be reordered.